usdevs/usc-website
# Updated usage of @fortawesome/react-fontawesome

# Pull Request (PR) - *Updated fontawesome usage*
## Updated dependencies and imports for @fortawesome/react-fontawesome

This PR fixes remaining issues with setting up the development environment. All changes taken with reference to [here](https://fontawesome.com/how-to-use/on-the-web/using-with/react).

### New dependencies in `package.json`
```
yarn add @fortawesome/fontawesome-svg-core
yarn add @fortawesome/free-solid-svg-icons
yarn add @fortawesome/react-fontawesome
```

### Imports
```
// Depreciated
import FontAwesomeIcon from '@fortawesome/react-fontawesome'

// Updated
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
```

### SVG library
```
// Depreciated
import fontawesome from '@fortawesome/fontawesome'
fontawesome.library.add( ... )

// Updated
import { library } from '@fortawesome/fontawesome-svg-core'
library.add( ... )
```

## Link To Issue 
"Urgent" fix, no issue opened. 

## How to test this change
1. Ensure node v10.13.0
2. Ensure yarn v1.2+
3. `yarn` and `yarn start` should succeed in deploying a local dev server

## Checklists
### Checklist for developer of feature
- [ x ] I have performed a code review on my own code
- [ ] All Unit Tests pass when ran
- [ x ] I have written an appropriate description for this PR.
- [ x ] I have written how to test this PR.

### Checklist for reviewers
